plugins/searchonline.koplugin/handleurlstring.lua

```diff
@@ -0,0 +1,171 @@
+local InfoMessage = require("ui/widget/infomessage")
+local UIManager = require("ui/uimanager")
+local url = require("socket.url")
+local util = require("util")
+local _ = require("gettext")
+
+local HandleUrlString = {}
+
+-- Message box to inform the user about a misformatted URL
+function HandleUrlString:urlError(text_explaining_error)
+    UIManager:show(InfoMessage:new{
+        text = _(text_explaining_error),
+        timeout = 3
+    })
+end
+
+-- Check if the URL is valid and show an error if not
+function HandleUrlString:isValidUrl(url_string)
+    local parsed_url = url.parse(url_string)
+
+    if not parsed_url then
+        self.urlError("Could not parse URL. It may be misformatted: " .. url_string)
+        return false
+    end
+
+    if not parsed_url.scheme then
+        self.urlError("The URL is missing http:/https: " .. url_string)
+        return false
+    end
+
+    if not parsed_url.host then
+        self.urlError("Missing or invalid host: " .. url_string)
+        return false
+    end
+
+    return true

```

Comment:
(That's my personal preference: we don't need that many blank lines, only between logical sections to help distinguising step. For such little simple functions, it makes it tedious to review to people with smaller screens than you.)